integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/util/SecondaryIndexUtil.scala

```diff
@@ -280,6 +286,35 @@ object SecondaryIndexUtil {
     }
   }
 
+  /**
+   * This method deletes the old index files or merge index file after data files merge
+   */
+  def deleteOldIndexOrMergeIndexFiles(factTimeStamp: Long,
+      validSegments: util.List[Segment],
+      indexCarbonTable: CarbonTable): Unit = {
+    // delete the index/merge index carbonFile of old data files
+    validSegments.asScala.foreach { segment =>
+      var indexFiles: util.Map[String, String] = null
+      // for old store scenario
+      if (segment.getSegmentFileName == null) {
+        val segmentPath = CarbonTablePath.getSegmentPath(indexCarbonTable.getTablePath,
+          segment.getSegmentNo)
+        indexFiles = new SegmentIndexFileStore().getMergeOrIndexFilesFromSegment(segmentPath)
+      } else {
+        val segmentFileStore = new SegmentFileStore(indexCarbonTable.getTablePath,
+          segment.getSegmentFileName)
+        indexFiles = segmentFileStore.getIndexOrMergeFiles
+      }
+      val indexFilesToDelete = indexFiles.keySet().asScala.filter { indexFile =>
+        DataFileUtil.getTimeStampFromFileName(indexFile).toLong >
+        factTimeStamp
+      }
+      indexFilesToDelete.foreach { indexFile =>
```

Comment:
no need to declare local variable indexFilesToDelete

Reply:
done